### PR TITLE
Fix JavaDoc syntax errors. Add check to circle build.

### PR DESCRIPTION
**Pull request by kaizimmerm**
It seems that the maven Javadoc Plugin update has now lint enabled by default. 

In order to have our release build run again I did:

* Reduced lint to syntax check only.
* Fixed many syntax errors in JavaDoc.
* Moved one class that is JPA specific out of the repository API package (i.e. it had JPA JavaDoc dependencies).
* Add JavaDoc check to Circle so that we can see errors in PRs.

While at it I updated EclipseLInk to 2.7.4

### CODE REVIEW

**Comment by schabdo** [comment]
@hawkbit-bot verify please

File: hawkbit-core/src/main/java/org/eclipse/hawkbit/repository/FieldNameProvider.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ default boolean containsSubEntityAttribute(final String propertyField) {
     /**
      * Is the entity field a {@link Map}.
      *
-     * @return <true> is a map <false> is not a map
+     * @return <code>true</code> is a map false is not a map
```
Comment (laverman) [inline hawkbit-core/src/main/java/org/eclipse/hawkbit/repository/FieldNameProvider.java]: surround 'false' also with <code> tag

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/CommonDialogWindow.java [hunk 2]
```diff
@@ -554,22 +554,23 @@ public void setCancelButtonEnabled(final boolean enabled) {
...
+         * @return <code>true</code> = window will close <code>false</code> =
+         *         will not closed.
          */
         default boolean canWindowClose() {
             return true;
         }
 
         /**
-         * Saves/Updates action. Is called if canWindowSaveOrUpdate is <true>.
+         * Saves/Updates action. Is called if canWindowSaveOrUpdate is true.
```
Comment (laverman) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/CommonDialogWindow.java]: Please add <code> tag to 'true'

**Comment by kaizimmerm** [comment]
@laverman good catch, done.

**Pull request merged**